open-craft/edx-custom-registration-form
# Add form and models for Phone Numbers

This PR contains a Registration Extension Form which adds Phone Number and Country Code fields to the User Registration Form
 
**JIRA tickets**: Implements OC-3375.

**Screenshots**:

**Testing instructions**:

1. Clone the repo to `devstack/src` directory: `cd devstack/src; git clone https://github.com/open-craft/edx-custom-registration-form.git`
2. Navigate inside vagrant devstack and install `cd /edx/src/edx-custom-registration-form; pip install -e .`
3. Edit lms.env.json:
    1. Find `REGISTRATION_EXTRA_FIELDS`, change `"city": "hidden"` to `"city": "required"`
    2. Ensure that `"country": "required"` is set. This was set by default for me.
    3. Add a new line above `REGISTRATION_EXTRA_FIELDS` containing: `"REGISTRATION_EXTENSION_FORM": "reg_form_ext.forms.PhoneInfoForm",`
    4. Add a new line containing: `"ADDL_INSTALLED_APPS": ["reg_form_ext"],`
    5. Save and exit editor
4. Restart the LMS by ending restarting in paver, or running `touch ~edxapp/edx-platform/manage.py`
5. Run migrations with `paver update_db`
6. Run tests with `paver test_system -s lms -t /edx/src/edx-custom-registration-form`


**Author notes and concerns**:

1. Including the country codes in `choices.py` is perhaps not the best place for them. I suspect we'd be better of sourcing these from a maintained module such as [django-phonenumber-field](https://github.com/stefanfoulis/django-phonenumber-field) or [python-phonenumbers](https://github.com/daviddrysdale/python-phonenumbers)

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
REGISTRATION_EXTRA_FIELDS:
  city: "required"
EDXAPP_EXTRA_REQUIREMENTS:
  - name: edx-custom-registration-form
    version: 1.0
  - name: git+https://github.com/open-craft/edx-custom-registration-form.git
EDXAPP_ENV_EXTRA:
  REGISTRATION_EXTENSION_FORM: "reg_form_ext.forms.PhoneInfoForm"
```
